cmd/satellite/main.go

```diff
@@ -63,50 +63,58 @@ func main() {
 		}
 	}()
 
-	go func() {
-		backlog, _ := recoverState(satellite_configuration.Satellite.Cache)
+	backlog, _ := recoverState(satellite_configuration.Satellite.Cache)
```

Comment:
The timestamp coming from the time the data is added to the DB seems like the root problem here to me. Is there a reason we don't timestamp the data packets?

Reply:
Because the satellites are all running on different machines, that might have different/out-of-sync clocks. By only taking the timestamp on entry in the db, we have one source of truth for timing.

Note for this same reason once we start displaying last seen times in the UI, I think the api endpoint should return the duration to when the last heartbeat occured, not the timestamp it occured on

Reply:
Backlog is there to as much as possible help there be as close as possible to no blindspots in our data. The idea is, if we fail between data uploads, and recover, then we can assume either one of two semantics:

1. Assume all data within the intervals was uncollectible, start streaming data from the present
2. Attempt to recover our state as much as possible, upload backlog, restart timer and stream from present.

Our state is actually quite small in the satellite, easily recoverable and therefore if we can help it, I think we'd like to ensure the groundstation has close enough to perfect data.

I'd think of it as a QoL feature, rather than an critical feature, but worth keeping.